contrib/src/contrib/uart_reader/__main__.py

```diff
@@ -0,0 +1,1700 @@
+            if len(port) == 0:
+                self.logger.error("No port selected, please select a valid port")
+                self.port_status.setText("Port Status: Invalid Port")
+                self.port_status.setStyleSheet("color: orange")
+                return
+            port = port[1][port[0]].split(" - ")[0]
+            baud = self.db.get_item("Serial Settings", "baud").value
+            baud = int(baud[1][baud[0]])
+            self.ser.try_start(port, baud)
+
+    # Serial Handlers
+    def connection_status_handler(self, status):
+        if status:
+            self.connect_disconnect_button.setText("Disconnect")
+            self.port_status.setText("Port Status: Connected")
+            self.port_status.setStyleSheet("color: green")
+        else:
+            self.connect_disconnect_button.setText("Connect")
+            self.port_status.setText("Port Status: Disconnected")
+            self.port_status.setStyleSheet("color: red")
+
+    def error_message_handler(self, message):
+        # self.logger.error(message)
+        pass
+
+    def normal_message_handler(self, message):
+        if message == "TERMINATE":
+            return
+        self.logger.info(f">>{message}")
+
+    def prefix_message_handler(self, prefix, message):
+        # self.logger.info(f"Received {len(message)} bytes for prefix {prefix}")
+        if prefix == self.db.get_item("Serial Settings", "database_prefix").value:
+            self.logger.info(f"New configuration received: {message}")
+        elif prefix == self.db.get_item("Audio Settings", "serial_prefix").value:
+            self.logger.info(f"New audio data received of length {len(message)}")
+            self.open_audio_window(message)
+            # Print the received data only to the log file
+            self.logger.debug(f"Received audio data: {message}")
+        elif prefix == self.db.get_item("MEL Settings", "serial_prefix").value:
+            self.logger.info(f"New MEL data received of length {len(message)}")
+            self.open_mel_window(message)
+            # Print the received data only to the log file
+            self.logger.debug(f"Received MEL data: {message}")
+        else:
+            self.logger.warning(
+                f"Unknown prefix {prefix} (message length: {len(message)})"
+            )
+
+
+####################################################################################################
+# Database Initialization
+####################################################################################################
+
+
+def database_init(db: dbu.ContentDatabase):
+    # Add the Hidden group
+    db.create_category("_hidden")
+    db.add_item(
+        "_hidden",
+        "appname",
+        db.ConstantText("App Name", "UART Reader", "The name of the application"),
+    )
+    db.add_item(
+        "_hidden",
+        "appversion",
+        db.ConstantText("App Version", "4.1", "The version of the application"),
```

Comment:
```suggestion
        db.ConstantText("App Version", __version__, "The version of the application"),
```

This is probably better to create a `__version__.py` file in `src/contrib/uart_reader.py` that has a version string `__version__ = "4.1"` inside, and you import the version at the top of `__main__.py`: `from .__version__ import __version__`.